pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -219,8 +232,10 @@ async def run(
                 with _logfire.span('preparing model and tools {run_step=}', run_step=run_step):
                     agent_model = await self._prepare_model(model_used, deps)
 
+                model_settings = {**(self.model_settings or {}), **(model_settings or {})}
```

Comment:
```suggestion
                # Note: we may want merge recursively if/when we add non-primitive values
                model_settings = {**(self.model_settings or {}), **(model_settings or {})}
```

Reply:
(Also see my comment about replacing this with a call to a new `pydantic_ai.settings.merge_settings` function

Reply:
I see Samuel had the same suggestion of pulling this out into a separate function